lib/defines.h

```diff
@@ -213,4 +213,13 @@
 #define PASS_MAX  BUFSIZ - 1
 #endif
 
+/* Maximum length of user name or group name
+ * This is used only to limit a string length passed to syslog.
+ * An issue "unbounded source buffer" is reported while doing 
+ * coverity scans, when there is an unlimited string length passed 
+ * to syslog function.
+ * Typical POSIX login name max length is 256.
+ */
+#define USER_ENTRY_MAX_LENGTH 256
```

Comment:
There is `LOGIN_NAME_MAX`, defined by POSIX in <limits.h>.

However, we support unlimited user-name lengths.  See this commit:

```
commit 6a1f45d932c859725bc175f96f7c42b9ccc20ea4
Author: Tobias Stoeckmann <tobias@stoeckmann.org>
Date:   Sat Feb 3 01:10:22 2024 +0100

    lib/chkname.c: Support unlimited user name lengths
    
    If the system does not have a user name length limit, support it
    accordingly. If the system has no _SC_LOGIN_NAME_MAX, use
    LOGIN_NAME_MAX constant instead.
    
    Signed-off-by: Tobias Stoeckmann <tobias@stoeckmann.org>
```

You should probably use a run-time variable user-name length.

You'll need to write this function (extracted from the current code in is_valid_user_name() from <lib/chkname.c>):

```c
size_t username_maxlen(void)
{
	long  maxlen;

	errno = 0;
	maxlen = sysconf(_SC_LOGIN_NAME_MAX);
	if (maxlen == -1 && errno != 0)
		return LOGIN_NAME_MAX;

	return maxlen;
}
```

and use it in your cases.

Reply:
BTW, now I remember we use the term size, not len, since it includes the NUL byte.

Reply:
Please call the newly added *login_name_max_size*() function instead.

Reply:
New approach is applied by direct usage of a new function: is_valid_user_name